src/Configuration/DatabaseParameters.php

```diff
@@ -0,0 +1,83 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Keboola\DbExtractorCommon\Configuration;
+
+class DatabaseParameters
+{
+    /** @var string */
+    private $host;
+
+    /** @var int|null */
+    private $port;
+
+    /** @var string */
+    private $database;
+
+    /** @var string */
+    private $user;
+
+    /** @var string */
+    private $password;
+
+    public function __construct(
+        string $host,
+        string $user,
+        string $password,
+        ?string $database = null,
+        ?int $port = null
+    ) {
+        $this->host = $host;
+        $this->port = $port;
+        $this->database = $database;
+        $this->user = $user;
+        $this->password = $password;
+    }
+
+    public function getHost(): string
+    {
+        return $this->host;
+    }
+
+    public function getPort(): ?int
+    {
+        return (int) $this->port;
+    }
+
+    public function getDatabase(): ?string
+    {
+        return $this->database;
+    }
+
+    public function getUser(): string
+    {
+        return $this->user;
+    }
+
+    public function getPassword(): string
+    {
+        return $this->password;
+    }
+
+    public function setHost(string $host): void
```

Comment:
Proč to má settery? Tohle je konfigurace, ta by měla být zas immutable. Co přidat (ale obecně si myslím, že by to měl dělat někdo jiný - ten kdo ví o tom tunelu - ten by měl rovnou udělat odpovídající credentials a ty by se neměly měnit. 

```
withSshTunnel(int $tunnelLocalPort) {
return new DatabaseParameters(
            '127.0.0.1',
            $this->getUser(),
            $this->getPassword(),
            $this->getDatabase(),
            $tunnelLocalPort
        );
} ```